src/wsdiscovery.ts

```diff
@@ -213,44 +306,59 @@ export default class WsDiscovery extends EventEmitter {
                         pid: this.networkDevicePID(name),
                     };
                     const device = new HuddlyDevice(deviceData);
-
-                    if (this.linkLocalAddrAllowed(device.ip.toString())) {
+                    if (this.isDeviceAllowed(device.ip.toString())) {
                         discoveredDevices.push(device);
                         this.emit('device', device);
                     } else {
                         Logger.debug(
-                            `Link local huddly ip camera detected. Instructions are set to ignore link local devices!`,
+                            `Ignoring device at ip address: ${device.ip.toString()}.`,
+                            WsDiscovery.name
+                        );
+                        Logger.debug(
+                            `Ignore local link devices: ${this.opts.ignoreLinkLocalDevices ||
+                                process.env.HUDDLY_WSDD_IGNORE_LINK_LOCAL_DEVICES}.`,
+                            WsDiscovery.name
+                        );
+                        Logger.debug(
+                            `Has full probing rights: ${this.opts.probeEntireNetwork ||
+                                process.env.PROBE_ENTIRE_NETWORK}`,
                             WsDiscovery.name
                         );
                     }
                 });
             }
             callback(discoveredDevices);
         };
-        this.socket.on('message', onProbeResponseHandler);
-        setTimeout(() => {
-            self.socket.removeListener('message', onProbeResponseHandler);
-            callback(discoveredDevices);
-        }, this.opts.timeout);
 
-        if (this.ifStateConnected) {
+        Object.values(this.socketConnections).forEach(socket => {
+            socket.on('message', onProbeResponseHandler);
+            setTimeout(() => {
+                socket.removeListener('message', onProbeResponseHandler);
+            }, this.opts.timeout);
+
             this.setTimeoutWithRandomDelay(
-                this.socket.send.bind(this.socket, body, 0, body.length, 3702, '239.255.255.250'),
+                socket.send.bind(socket, body, 0, body.length, 3702, '239.255.255.250'),
                 this.maxDelay
             );
-        }
+        });
+
+        setTimeout(() => {
+            callback(discoveredDevices);
+        }, this.opts.timeout);
```

Comment:
I think this might be a problem if the socket gets a `on.('message')` event within the foreach loop which invokes the `onProbeResponseHandler` which in turn calls `callback(discoveredDevices);`. Here you are calling the callback again. 

I might be wrong but maybe it is a better idea to deal with the timeouts only inside the loop?

```
const probeTimeout = setTimeout(() => {
    socket.removeListener('message', onProbeResponseHandler);
}, this.opts.timeout);

socket.on('message', (message) => {
   onProbeResponseHandler(message);
   clearTimeout(probeTimeout);
});

this.setTimeoutWithRandomDelay(
    socket.send.bind(socket, body, 0, body.length, 3702, '239.255.255.250'),
    this.maxDelay
);
```
I did not do this before because we only initialized one socket, whereas now you have potentially a list of sockets you will be sending messages to. What are your thoughts about this?

Reply:
Yes, it's probably better to just remove it